apps/examples/stack_protection/stack_protection_main.c

```diff
@@ -26,12 +26,38 @@
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
-#define	STACK_NORMAL_SIZE		10
+#define	STACK_NORMAL_SIZE		300
 #define	STACK_OVERFLOW_SIZE		4096
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 
+void stack_prot_overflow(int *val, char ch)
+{
+	char *ptr;
+	int i = 0;
+	char array_normal[STACK_NORMAL_SIZE];
+
+	ptr = &array_normal[STACK_NORMAL_SIZE];
+	for (i = 0; i < STACK_NORMAL_SIZE; i++) {
+		*ptr = 'a';
+		ptr = ptr - 1;
+	}
+	printf("Array addr = 0x%x\n", &array_normal[0]);
+
+	*val = *val + 1;
+	if (ch == 'N' || ch == 'n') {
+		if (*val == ((STACK_OVERFLOW_SIZE / STACK_NORMAL_SIZE) / 2))
+			return;
+	}
+	else {
+		if (*val == (STACK_OVERFLOW_SIZE / STACK_NORMAL_SIZE))
+			return;
+	}
+
+	stack_prot_overflow(val, ch);
```

Comment:
Stack limit checking happens when value of SP goes beyond SPLIM. 
This can happen in 2 cases - recursion and definition of large arrays. 
I think it would be good to test both these scenarios.

Reply:
added both situation for test case for overflow condition check